### PR TITLE
InaSAFE Profile

### PR DESCRIPTION
**Pull request by ismailsunni**
### What does it fix?
* Ticket: #4482 
* Funded by: DFAT
* Description: 
   - Create widget to edit InaSAFE profile, add it to options.
   - Use the InaSAFE Profile to get displacement rate and affected
inasafe_profile

### Checklist:
- [x] Commit messages are descriptive and explain the rationale for changes
- [ ] Add to the changelog in metadata.txt if it's a new feature
- [x] Unit test for new code added
- [x] Request someone to review or test your PR

### CODE REVIEW

**Comment by ismailsunni** [comment]
@Gustry 
Can you check especially how I use the keyword parameter in `post_processor_displaced_ratio`? I am not sure it will get dictionary or string.

**Comment by ismailsunni** [comment]
Ouch, circular import :(

**Comment by ismailsunni** [comment]
@Gustry 
I revert my commit for adding the `post_processor_displaced_ratio`, since I got circular import. It's tricky to fix.

File: safe/definitions/test/test_utilities.py [hunk 1]
```diff
@@ -511,6 +528,104 @@ def test_get_provenance(self):
...
+        default_profile = generate_default_profile()
+        # from pprint import pprint
+        # pprint(default_profile)
+        self.assertIsInstance(default_profile, dict)
+        self.assertIsNotNone(default_profile)
+
+    def test_get_displacement_rate_and_affected(self):
+        """Test for get_displacement_rate and is_affected"""
+        # Random key
+        value = is_affected('foo', 'bar', 'boom')
```
Comment (Gustry) [inline safe/definitions/test/test_utilities.py]: Why not nagasari?

File: safe/gui/tools/options_dialog.py [hunk 2]
```diff
@@ -121,7 +126,20 @@ def __init__(self, iface, parent=None, qsetting=''):
         self.grpNotImplemented.hide()
         self.adjustSize()
 
+        # Profile Preference Tab
+        self.preference_label = QLabel()
+        self.preference_label.setText(tr(
+            'Please set your affected and displacement rate for each hazard, '
+            'classification, and its class below. The value of displacement '
+            'rate is from 1% to 100%.'
```
Comment (Gustry) [inline safe/gui/tools/options_dialog.py]: not 0%?

File: safe/gui/widgets/profile_widget.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+                    data[hazard][classification][the_class] = {
+                        'affected': affected_check_box.isChecked(),
+                        'displacement_rate': displacement_rate_spin_box.value()
+                    }
+        return data
+
+
+class PercentageSpinBox(QDoubleSpinBox):
+
+    """Custom Spinbox for percentage 0 % - 100 %."""
```
Comment (Gustry) [inline safe/gui/widgets/profile_widget.py]: So here it's 0%

File: safe/gui/widgets/profile_widget.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+__copyright__ = "Copyright 2017, The InaSAFE Project"
+__license__ = "GPL version 3"
+__email__ = "info@inasafe.org"
+__revision__ = '$Format:%H$'
+
+
+class ProfileWidget(QTreeWidget, object):
+
+    """Profile Widget"""
```
Comment (Gustry) [inline safe/gui/widgets/profile_widget.py]: My regex will fix that, I don't care now ;-)

File: safe/definitions/test/test_utilities.py (hunk 1 above)
Comment (ismailsunni) [reply safe/definitions/test/test_utilities.py]: Nagasari is too long to write :D

File: safe/gui/tools/options_dialog.py (hunk 2 above)
Comment (ismailsunni) [reply safe/gui/tools/options_dialog.py]: Fixed.

File: safe/gui/widgets/profile_widget.py (hunk 4 above)
Comment (ismailsunni) [reply safe/gui/widgets/profile_widget.py]: Yeay.

File: safe/gui/widgets/profile_widget.py (hunk 3 above)
Comment (ismailsunni) [reply safe/gui/widgets/profile_widget.py]: Yes, this is the correct one.

**Comment by ismailsunni** [comment]
I am merging this one first, before it becomes bigger PR. It hasn't been used in the post processor. It seems I need to re-structure the directory.

**Pull request merged**